libraries/lib-module-manager/PluginManager.cpp

```diff
@@ -333,6 +338,29 @@ PluginManager::~PluginManager()
    Terminate();
 }
 
+void PluginManager::InitializePlugins()
+{
+   ModuleManager & mm = ModuleManager::Get();
+
+   // Check all known plugins to ensure they are still valid.
+   for (auto &pair : mRegisteredPlugins) {
+      auto &plug = pair.second;
+      const wxString & plugPath = plug.GetPath();
+      PluginType plugType = plug.GetPluginType();
+      
+      if (plugType != PluginTypeNone && plugType != PluginTypeStub && plugType != PluginTypeModule)
+      {
+         plug.SetValid(mm.IsPluginValid(plug.GetProviderID(), plugPath, true));
```

Comment:
Moving the definition makes it clearer that false became true in this line.  A more complete check of validity is done.

Still now, I wonder about this.  What happens if a plug-in is upgraded to a newer version reinstalled at the same path?  Should that be detected and cause another load again in the sandbox?

Nyquist.cpp contains checks for advance of the modification date of the plug-in file, making it easier to try changes while you develop a plug-in.  But those are just interpreted source code.

Maybe recording of time stamps is a future enhancement to do.

Reply:
>Still now, I wonder about this. What happens if a plug-in is upgraded to a newer version reinstalled at the same path? Should that be detected and cause another load again in the sandbox?

I'd say yes. But for this we need to track module file timestamp (if there is a file)